pages/api/auth/updatePassword.ts

```diff
@@ -2,61 +2,65 @@ import { NextApiRequest, NextApiResponse } from "next";
 import { getErrorMessage } from "@/lib/api/apiHelpers";
 import { db } from "@/lib/api/db";
 import {
-    USER_ID,
-    checkIfPasswordHasBeenSet,
+    checkIfAdminExists,
     hashPassword,
 } from "@/lib/api/auth/authHelpers";
 import { compare } from "bcryptjs";
 import { getApiError } from "@/lib/api/error";
+import { getToken } from "next-auth/jwt";
 
 export default async function updatePasswordAPI(
     req: NextApiRequest,
     res: NextApiResponse
 ) {
-    throw new Error("TODO: Implement updatePasswordAPI")
-    // try {
-    //     const jsonData = req.body;
+    try {
+        const token = await getToken({req : req});
+        if (!token || !token.name){
+            throw new Error("You must log in first");
+        }
+        const username = token.name;
+        const jsonData = req.body;
 
-    //     const passwordHasBeenSet = await checkIfPasswordHasBeenSet();
-    //     if (!passwordHasBeenSet) {
-    //         throw new Error("Must go through the global setup process first");
-    //     }
-    //     const newPassword = await hashPassword(jsonData.newPassword);
-    //     const oldPassword = jsonData.oldPassword;
+        const passwordHasBeenSet = await checkIfAdminExists();
+        if (!passwordHasBeenSet) {
+            throw new Error("Must go through the global setup process first");
+        }
+        const newPassword = await hashPassword(jsonData.newPassword);
+        const oldPassword = jsonData.oldPassword;
 
-    //     const currentDate = new Date(Date.now());
-    //     const userInDB = await db.user.findFirst({
-    //         where: {
-    //             username: USER_ID,
-    //         },
-    //     });
+        const userInDB = await db.user.findFirst({
```

Comment:
At some point before this we should check whether we're actually logged in as the user we're trying to update. Or we're an admin

Reply:
in this api the username comes from their token -  I'll add an api for admins to update someone else's password